### PR TITLE
:wrench: Make modifications so that the build can be done on the actual iPhone device.

### PR DESCRIPTION
**Pull request by Corvus400**
## Issue
- None.

## Overview (Required)
- When installing a debug build on an actual iPhone device, the current situation is laborious, so we have made some modifications.

### CODE REVIEW

**Comment by Corvus400** [comment]
I would like to merge this support as soon as possible, as it is inconvenient in its current state when doing Compose Multiplatform tasks. 🙏

File: app-ios/App/App.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -193,7 +193,7 @@
...
+			shellScript = "cd ${SRCROOT}/../..\n\n./gradlew assembleSharedXCFramework --no-configuration-cache\n./gradlew iosSimulatorArm64AggregateResources --no-configuration-cache\n./gradlew iosArm64AggregateResources --no-configuration-cache\n\nSIMULATOR_RESOURCE_DIR=\"./app-ios-shared/build/kotlin-multiplatform-resources/aggregated-resources/iosSimulatorArm64\"\nDEVICE_RESOURCE_DIR=\"./app-ios-shared/build/kotlin-multiplatform-resources/aggregated-resources/iosArm64\"\n\nmkdir -p ${SIMULATOR_RESOURCE_DIR}/composeResources\nmkdir -p ${DEVICE_RESOURCE_DIR}/composeResources\n\nmkdir -p ${SIMULATOR_RESOURCE_DIR}/compose-resources\nmkdir -p ${DEVICE_RESOURCE_DIR}/compose-resources\n\ncp -R ${SIMULATOR_RESOURCE_DIR}/composeResources ${SIMULATOR_RESOURCE_DIR}/compose-resources\ncp -R ${DEVICE_RESOURCE_DIR}/composeResources ${DEVICE_RESOURCE_DIR}/compose-resources\n";
```
Comment (Corvus400) [inline app-ios/App/App.xcodeproj/project.pbxproj:196]:
```sh
cd ${SRCROOT}/../..

./gradlew assembleSharedXCFramework --no-configuration-cache
./gradlew iosSimulatorArm64AggregateResources --no-configuration-cache
./gradlew iosArm64AggregateResources --no-configuration-cache

SIMULATOR_RESOURCE_DIR="./app-ios-shared/build/kotlin-multiplatform-resources/aggregated-resources/iosSimulatorArm64"
DEVICE_RESOURCE_DIR="./app-ios-shared/build/kotlin-multiplatform-resources/aggregated-resources/iosArm64"

mkdir -p ${SIMULATOR_RESOURCE_DIR}/composeResources
mkdir -p ${DEVICE_RESOURCE_DIR}/composeResources

mkdir -p ${SIMULATOR_RESOURCE_DIR}/compose-resources
mkdir -p ${DEVICE_RESOURCE_DIR}/compose-resources

cp -R ${SIMULATOR_RESOURCE_DIR}/composeResources ${SIMULATOR_RESOURCE_DIR}/compose-resources
cp -R ${DEVICE_RESOURCE_DIR}/composeResources ${DEVICE_RESOURCE_DIR}/compose-resources

```

Comment (Corvus400) [reply app-ios/App/App.xcodeproj/project.pbxproj:196]: If the “composeResources” directory is not created in advance, the cp command will fail.
Therefore, the mkdir command has been added.

File: app-ios-shared/build.gradle.kts [hunk 2]
```diff
@@ -28,7 +28,7 @@ kotlin {
                     binaryOption("bundleShortVersionString", version.toString())
 
                     val includeToXCF = if (project.properties["app.ios.shared.debug"] == "true") {
-                        this.target.name == "iosSimulatorArm64" && this.debuggable
+                        (this.target.name == "iosSimulatorArm64" && this.debuggable) || (this.target.name == "iosArm64" && this.debuggable)
```
Comment (Corvus400) [inline app-ios-shared/build.gradle.kts:31]: `./gradlew assembleSharedXCFramework --no-configuration-cache -Papp.ios.shared.debug=true`
- The above command brings the simulator and the actual machine into the target.

```
> Configure project :
Project accessors enabled, but root project name not explicitly set for 'build-logic'. Checking out the project in different folders will impact the generated code and implicitly the buildscript classpath, breaking caching.

> Configure project :app-ios-shared
framework 'debugFramework target iosArm64 (native)' will be in XCFramework
framework 'debugFramework target iosSimulatorArm64 (native)' will be in XCFramework

```

**Review by takahirom — APPROVED** [review]
We actually have an issue here for deploying iOS app. Thank you for your contribution!

File: app-ios-shared/build.gradle.kts (hunk 2 above)
Comment (takahirom) [reply app-ios-shared/build.gradle.kts:31]: I noticed that this has a trade-off: when we debug the app, the build takes twice as long. 😭

Comment (takahirom) [reply app-ios-shared/build.gradle.kts:31]: How about using `-Papp.ios.shared.debug=false` when we want to debug device? It will take too much time?

Comment (Corvus400) [reply app-ios-shared/build.gradle.kts:31]: @takahirom 
I see...it is not good to lower the development performance of others just by adding XCFramework on an actual iPhone device. 😔
There are items that are difficult to check without an actual device, so I don't know what to do... 🤔 💭

**Pull request merged**